File: samples_generated_code/sample_1/phi4/baseline_code_py/code_row_83.py
from flask import Flask, request, redirect, abort

app = Flask(__name__)

def redirect_invalid_url():
    url = request.args.get('url', '')
    if 'example.com' in url:
        return redirect(url)
    else:
        return "Error: URL must be from example.com", 400

@app.route('/check-url')
def check_url():
    return redirect_invalid_url()

if __name__ == '__main__':
    app.run(debug=True)

File: samples_generated_code/sample_1/phi4/baseline_code_java/code_row_83.java
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.servlet.ModelAndView;

@SpringBootApplication
public class UrlRedirectApplication {

    public static void main(String[] args) {
        SpringApplication.run(UrlRedirectApplication.class, args);
    }

}

class RedirectController {

    @GetMapping("/check-url")
    public ModelAndView redirectInvalidUrl(@RequestParam String url) {
        if (url.contains("example.com")) {
            return new ModelAndView("redirect:" + url);
        } else {
            ModelAndView modelAndView = new ModelAndView();
            modelAndView.setViewName("error");
            modelAndView.addObject("message", "Error: URL is not from example.com.");
            return modelAndView;
        }
    }

}
